# a. scan context:
scan_context_path: /slam_data/scan_context   

# 回环检测:
loop_closure_method: scan_context # 选择回环检测方法, 目前支持scan_context

# 匹配
registration_method: NDT_CPU   # 选择点云匹配方法，目前支持 ICP_SVD , ICP , NDT , NDT_CPU , SICP , NDT_CUDA , ICP_CUDA , NDT_GPU

# 当前帧
# no_filter指不对点云滤波，在匹配中，理论上点云越稠密，精度越高，但是速度也越慢
# 所以提供这种不滤波的模式做为对比，以方便使用者去体会精度和效率随稠密度的变化关系
frame_filter: voxel_filter # 选择当前帧点云滤波方法，目前支持：voxel_filter、no_filter

# 局部地图
# 局部地图从全局地图切割得到，此处box_filter_size是切割区间
# 参数顺序是min_x, max_x, min_y, max_y, min_z, max_z
box_filter_size: [-150.0, 150.0, -150.0, 150.0, -150.0, 150.0]
# box_filter_size: [-1.0, 1.0, -1.0, 1.0, -1.0, 1.0]

local_map_filter: voxel_filter # 选择滑窗地图点云滤波方法，目前支持：voxel_filter、no_filter

# 全局地图
global_map_filter: voxel_filter # 选择滑窗地图点云滤波方法，目前支持：voxel_filter、no_filter


# 各配置选项对应参数
## ScanContext params:
scan_context:
    # a. ROI definition:
    max_radius: 80.0
    max_theta: 360.0
    # b. resolution:
    num_rings: 20
    num_sectors: 60
    # c. ring key indexing interval:
    indexing_interval: 1
    # d. min. key frame sequence distance:
    min_key_frame_seq_distance: 100
    # e. num. of nearest-neighbor candidates to check:
    num_candidates: 5
    # f. sector key fast alignment search ratio:
    #   avoid brute-force match using sector key
    fast_alignment_search_ratio: 0.1
    # g. scan context distance threshold for proposal generation:
    #   0.4-0.6 is good choice for using with robust kernel (e.g., Cauchy, DCS) + icp fitness threshold 
    #   if not, recommend 0.1-0.15
    scan_context_distance_thresh: 0.15
## 匹配相关参数
## 匹配相关参数
ICP:
    max_corr_dist : 1.2
    trans_eps : 0.01
    euc_fitness_eps : 0.36
    max_iter : 30
ICP_SVD:
    max_corr_dist : 0.5
    trans_eps : 0.001
    euc_fitness_eps : 0.36
    max_iter : 280
ICP_CUDA:
    max_corr_dist : 1.2
    trans_eps : 0.01
    euc_fitness_eps : 0.36
    max_iter : 30
NDT:
    res : 0.5                 # volex  resolution
    step_size : 0.1    # 梯度下降的步长，越大下降越快，但是容易over shoot陷入局部最优
    trans_eps : 0.005    # 最大容差，一旦两次转换矩阵小于 trans_eps  退出迭代
    max_iter : 130         #   最大迭代次数
NDT_CPU:
    res : 0.5                 # volex  resolution
    step_size : 0.1    # 梯度下降的步长，越大下降越快，但是容易over shoot陷入局部最优
    trans_eps : 0.005    # 最大容差，一旦两次转换矩阵小于 trans_eps  退出迭代
    max_iter : 80         #   最大迭代次数
NDT_GPU:
    res : 0.5                 # volex  resolution
    step_size : 0.1    # 梯度下降的步长，越大下降越快，但是容易over shoot陷入局部最优
    trans_eps : 0.005    # 最大容差，一旦两次转换矩阵小于 trans_eps  退出迭代
    max_iter : 120         #   最大迭代次数
SICP:
    p : 1.0
    mu : 10.0
    alpha : 1.2
    max_mu : 1e5
    max_icp : 100
    max_outer : 100
    max_inner : 1
    stop : 1e-5
NDT_CUDA:
    res : 0.5                 # volex  resolution
    step_size : 0.1    # 梯度下降的步长，越大下降越快，但是容易over shoot陷入局部最优
    trans_eps : 0.005    # 最大容差，一旦两次转换矩阵小于 trans_eps  退出迭代
    max_iter : 200         #   最大迭代次数
## 滤波相关参数
voxel_filter:
    global_map:
        leaf_size: [0.15, 0.15, 0.15]
    local_map:
        leaf_size: [0.15, 0.15, 0.15]
    frame:
        leaf_size: [0.02, 0.02, 0.02]
